test/test_list.cpp

```diff
@@ -0,0 +1,716 @@
+        CHECK(!sum->is_null());
+        CHECK_EQUAL(ret_count, 2000);
+        CHECK_EQUAL(sum->template get<ColumnSumType<T>>(), ColumnSumType<T>(-1000));
+        auto avg = lst.avg(&ret_count);
+        CHECK(avg);
+        CHECK(!avg->is_null());
+        CHECK_EQUAL(ret_count, 2000);
+        CHECK_EQUAL(avg->template get<ColumnAverageType<T>>(),
+                    (ColumnAverageType<T>(-1000) / ColumnAverageType<T>(2000)));
+    }
+
+    lst.clear();
+    CHECK_EQUAL(lst.size(), 0);
+    {
+        size_t ret_ndx = realm::npos;
+        auto min = lst.min(&ret_ndx);
+        CHECK(min);
+        CHECK_EQUAL(ret_ndx, realm::npos);
+        ret_ndx = realm::npos;
+        auto max = lst.max(&ret_ndx);
+        CHECK(max);
+        CHECK_EQUAL(ret_ndx, realm::npos);
+        size_t ret_count = realm::npos;
+        auto sum = lst.sum(&ret_count);
+        CHECK(sum);
+        CHECK_EQUAL(ret_count, 0);
+        ret_count = realm::npos;
+        auto avg = lst.avg(&ret_count);
+        CHECK(avg);
+        CHECK_EQUAL(ret_count, 0);
+    }
+
+    lst.add(T(1));
+    {
+        size_t ret_ndx = realm::npos;
+        auto min = lst.min(&ret_ndx);
+        CHECK(min);
+        CHECK(!min->is_null());
+        CHECK_EQUAL(ret_ndx, 0);
+        CHECK_EQUAL(min->template get<ColumnMinMaxType<T>>(), ColumnMinMaxType<T>(1));
+        auto max = lst.max(&ret_ndx);
+        CHECK(max);
+        CHECK(!max->is_null());
+        CHECK_EQUAL(ret_ndx, 0);
+        CHECK_EQUAL(max->template get<ColumnMinMaxType<T>>(), ColumnMinMaxType<T>(1));
+        size_t ret_count = 0;
+        auto sum = lst.sum(&ret_count);
+        CHECK(sum);
+        CHECK(!sum->is_null());
+        CHECK_EQUAL(ret_count, 1);
+        CHECK_EQUAL(sum->template get<ColumnSumType<T>>(), ColumnSumType<T>(1));
+        auto avg = lst.avg(&ret_count);
+        CHECK(avg);
+        CHECK(!avg->is_null());
+        CHECK_EQUAL(ret_count, 1);
+        CHECK_EQUAL(avg->template get<ColumnAverageType<T>>(), ColumnAverageType<T>(1));
+    }
+
+    t->rollback();
+}
+
+TEST(List_AggOps)
+{
+    SHARED_GROUP_TEST_PATH(path);
+
+    std::unique_ptr<Replication> hist(make_in_realm_history());
+    DBRef sg = DB::create(*hist, path, DBOptions(crypt_key()));
+
+    test_lists_numeric_agg<int64_t>(test_context, sg, type_Int);
+    test_lists_numeric_agg<float>(test_context, sg, type_Float);
+    test_lists_numeric_agg<double>(test_context, sg, type_Double);
+    test_lists_numeric_agg<Decimal128>(test_context, sg, type_Decimal);
+
+    test_lists_numeric_agg<Optional<int64_t>>(test_context, sg, type_Int, Optional<int64_t>{}, true);
+    test_lists_numeric_agg<float>(test_context, sg, type_Float, realm::null::get_null_float<float>(), true);
+    test_lists_numeric_agg<double>(test_context, sg, type_Double, realm::null::get_null_float<double>(), true);
+    test_lists_numeric_agg<Decimal128>(test_context, sg, type_Decimal, Decimal128(realm::null()), true);
+}
+
+TEST(List_NestedListColumns)
```

Comment:
Just a general comment we are missing some testing around links since you have added support for them compared to the original implementation.

Reply:
This PR soes not include support for links